src/daft-micropartition/src/column_stats/from_parquet.rs

```diff
@@ -0,0 +1,425 @@
+use arrow2::array::PrimitiveArray;
+use daft_core::{
+    datatypes::{
+        logical::{DateArray, Decimal128Array, TimestampArray},
+        BinaryArray, BooleanArray, Int128Array, Int32Array, Int64Array, Utf8Array,
+    },
+    IntoSeries, Series,
+};
+use parquet2::{
+    schema::types::{PhysicalType, PrimitiveConvertedType, TimeUnit},
+    statistics::{
+        BinaryStatistics, BooleanStatistics, FixedLenStatistics, PrimitiveStatistics, Statistics,
+    },
+};
+use snafu::{OptionExt, ResultExt};
+
+use super::ColumnRangeStatistics;
+use crate::column_stats::MissingParquetColumnStatisticsSnafu;
+
+use crate::column_stats::UnableToParseUtf8FromBinarySnafu;
+use crate::utils::deserialize::convert_i96_to_i64_timestamp;
+
+impl TryFrom<&BooleanStatistics> for ColumnRangeStatistics {
+    type Error = super::Error;
+    fn try_from(value: &BooleanStatistics) -> Result<Self, Self::Error> {
+        if let Some(lower) = value.min_value && let Some(upper) = value.max_value {
+            ColumnRangeStatistics::new(
+                Some(BooleanArray::from(("lower", [lower].as_slice())).into_series()),
+                Some(BooleanArray::from(("upper", [upper].as_slice())).into_series()),
+            )
+        } else {
+            Ok(ColumnRangeStatistics::Missing)
+        }
+    }
+}
+
+impl TryFrom<&BinaryStatistics> for ColumnRangeStatistics {
+    type Error = super::Error;
+
+    fn try_from(value: &BinaryStatistics) -> Result<Self, Self::Error> {
+        if value.min_value.is_none() || value.max_value.is_none() {
+            return Ok(ColumnRangeStatistics::Missing);
+        }
+
+        let lower = value
+            .min_value
+            .as_ref()
+            .context(MissingParquetColumnStatisticsSnafu)?;
+        let upper = value
+            .max_value
+            .as_ref()
+            .context(MissingParquetColumnStatisticsSnafu)?;
+        let ptype = &value.primitive_type;
+
+        if let Some(ltype) = ptype.logical_type {
+            use parquet2::schema::types::PrimitiveLogicalType;
+            match ltype {
+                PrimitiveLogicalType::String
+                | PrimitiveLogicalType::Enum
+                | PrimitiveLogicalType::Uuid
+                | PrimitiveLogicalType::Json => {
+                    let lower = String::from_utf8(lower.clone())
+                        .context(UnableToParseUtf8FromBinarySnafu)?;
+                    let upper = String::from_utf8(upper.clone())
+                        .context(UnableToParseUtf8FromBinarySnafu)?;
+
+                    let lower =
+                        Utf8Array::from(("lower", [lower.as_str()].as_slice())).into_series();
+                    let upper =
+                        Utf8Array::from(("upper", [upper.as_str()].as_slice())).into_series();
```

Comment:
Oh just realized too that if we ever do any arithmetic on the series, they might get really weird with the names if we don't slot them in the correct spots (e.g. `lower + upper` will be called `lower`)

Doesn't really matter though?